backend/data.py

```diff
@@ -0,0 +1,84 @@
+from flask import request
+from flask import jsonify
+from flask import current_app
+from backend.db import *
+
+app = current_app
+
+@app.route("/getcommit", methods=("GET","POST"))
+def retrieve_commit_data():
+    db = get_db()
+
+    if request.method == "POST":
+        content_type = request.headers["Content-type"]
+        charset = request.headers["charset"]
+
+        if content_type != "application/json":
+            return
+        
+        commit = request.get_json()["commits"][0]
+        parent_commit = request.get_json["commits"][1]
+        db.execute("insert into commitinfo value({},{},{})".format(commit["commit_id"], commit["timestamp"], parent_commit["parent_commit_id"]))
+
+        for item in commit["code_size"]:
+            db.execute("insert into filechange value({}, {},{},{},{},{})",
+                        commit["commit_id"], 
+                        item.key(), 
+                        int(item.value()["text"]), 
+                        int(item.value()["data"]),
+                        int(item.value()["bss"]),
+                        int(item.value()["total"]))
+
+        # 若此条commit的上一条commit不在数据库中，插入这一条commit的记录。
+        if db.execute("select * from commitinfo where id = ?", (parent_commit["parent_commit_id"],), ).fetchall() is None:
```

Comment:
What about returning some false info with the latest commit id in database so that github action can send the missing commits info between the latest commit on github and the one in database.

Reply:
因为之前是假定从新建一个repo的第一条commit开始用这个插件，所有的commit都在这里面。但是后面考虑到这个插件可能是中途开始用（也就是之前有了若干commit，但是没有在我们的数据库当中），因此就可能需要补上这些commit。